RPC frontend (#1406) doesn't compile on macOS

Just pulled 5c58816 and it's not building. I'm assuming this is macOS specific.

```
[ 13%] Building frontends/rpc/rpc_frontend.o
frontends/rpc/rpc_frontend.cc:520:81: error: use of undeclared identifier 'environ'
                        if (posix_spawnp(&pid, argv[0], file_actions_p, /*attrp=*/NULL, argv.data(), environ) != 0) {
                                                                                                     ^
1 error generated.
make: *** [frontends/rpc/rpc_frontend.o] Error 1
```

--- a/frontends/rpc/rpc_frontend.cc
+++ b/frontends/rpc/rpc_frontend.cc
@@ -34,6 +34,8 @@
 #include "libs/sha1/sha1.h"
 #include "kernel/yosys.h"
 
+extern char **environ;
+
 YOSYS_NAMESPACE_BEGIN
 
 #if defined(_WIN32)

